src/main/java/com/marklogic/kafka/connect/source/CsvPlanInvoker.java

```diff
@@ -29,9 +33,12 @@ public Results invokePlan(PlanBuilder.Plan plan, String topic) {
         if (result.get() != null) {
             try (BufferedReader reader = new BufferedReader(new StringReader(result.get()))) {
                 String headers = reader.readLine();
+                AtomicLong rowNumber = new AtomicLong(0);
                 reader.lines().forEach(line -> {
+                    KeyGenerator keyGenerator = KeyGenerator.newKeyGenerator(parsedConfig, rowNumber.getAndIncrement());
+                    String key = keyGenerator != null ? keyGenerator.getUniqueKey() : null;
```

Comment:
The KeyGenerator instance should be an instance field.
Then, the call to getUniqueKey should accept the parameters that are unique to the row, and generate the unique key based on them.
That way the KeyGenerator instance will be reusable throughout the life of the Invoker.

Reply:
I think we only want it to be reused throughout a single `invokePlan`. While the UUID one could be reused throughout the life of the invoker, the Timestamp one cannot be since it includes the row number. And for each `invokePlan` call, we'd want the first row to have a row number of 1.